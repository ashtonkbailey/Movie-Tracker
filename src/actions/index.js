// movie actions
/////// add movies
export const addMovies = (movies) => ({
  type: 'ADD_MOVIES',
  movies
})

// user actions
//////// add favorite, delete favorite, sign in, sign out, create user

// export const addUser = (userObject) => ({
//   type: 'ADD_USER',
//   user
// })

// export const logInUser = (userObject) => ({
//   type: 'LOG_IN_USER',
//   user
// })