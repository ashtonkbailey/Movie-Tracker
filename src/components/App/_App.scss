.App {
  text-align: center;
  height: 100vh;
}

h1 {
  background: url(./assets/marquee.png)
    no-repeat
    top center / 300px 200px
    fixed;
  margin: auto;
  height: 200px;
  padding-top: 1.3em;
  font-size: 2.5em;
  color: rgba(254, 235, 158, 1);
  text-shadow:
    0 0 5px rgba(254, 235, 158, 1),
    0 0 0px rgba(254, 235, 158, 1),
    0 0 10px rgba(254, 235, 158, 1);
}

