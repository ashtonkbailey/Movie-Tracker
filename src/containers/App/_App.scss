.App {
  height: 100vh;
}

.title {
  background: url(./assets/marquee.png)
    no-repeat
    top center / 300px 200px;
}

h1 {
  text-align: center;
  margin: auto;
  height: 160px;
  padding-top: 1.2em;
  font-size: 2.5em;
  color: rgba(254, 235, 158, 1);
  font-family: 'Sanchez', serif;
  text-shadow:
    0 0 5px rgba(254, 235, 158, 1),
    0 0 0px rgba(254, 235, 158, 1),
    0 0 10px rgba(254, 235, 158, 1);
}

a {
  text-decoration: none;
}

.go-home {
  font-family: 'Sanchez', serif;
  font-size: 1.5rem;
  display: block;
  margin: 0 auto 30px;
  width: 250px;
  height: 50px;
  background: #950202;
  border: none;
  border-radius: 2px;
  color: #fff;
  transition: .4s ease;
  cursor: url('./assets/finger.png'), auto;
  outline: none;
  box-shadow: 0 0 6px 1px rgba(0, 0, 0, 0.5),
        inset 0 0 6px 1px rgba(0, 0, 0, 0.4);

  &:hover {
    color: rgb(255, 216, 58);
    font-size: 1.65rem;
    text-shadow: 0 0 3px rgba(0, 0, 0, 0.4);
  }
}