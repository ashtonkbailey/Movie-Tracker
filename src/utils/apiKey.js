export const key = '621a2082a320a64dd1c723f247130424'