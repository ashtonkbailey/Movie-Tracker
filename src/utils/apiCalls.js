import { key } from './apiKey'

export const fetchMovies = async() => {
  const url = `https://api.themoviedb.org/3/movie/now_playing?api_key=${key}&language=en-US&page=1`;
  const results = await fetch(url);
  const movieData = await results.json();
  return movieData.results
}

export const addNewUserFetch = async (user) => {
  const url = 'http://localhost:3000/api/users/new'
  const result = await fetch(url, {
    method: 'POST',
    body: JSON.stringify(user),
    headers: {
      'Content-Type': 'application/json'
    }
  })
  if (!result.ok) {
    throw new Error('User already exists!')
  }
  const data = await result.json()
  return data.id
}

export const signInUser = async (user) => {
  const url = 'http://localhost:3000/api/users'
  const result = await fetch(url, {
    method: 'POST',
    body: JSON.stringify(user),
    headers: {
      'Content-Type': 'application/json'
    }
  })
  if (!result.ok) {
    throw new Error('Email or password do not match!')
  }
  const returnedUsers = await result.json()
  return returnedUsers.data
}

