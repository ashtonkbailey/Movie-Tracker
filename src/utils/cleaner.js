import { fetchMovies } from './apiCalls';

export const cleanMovies = async()=> {
  const uncleanMovies = await fetchMovies()
  const cleanMovies = uncleanMovies.map(movie =>({
      title: movie.title,
      id: movie.id,
      rating: movie.vote_average,
      text: movie.overview,
      release: movie.release_date,
      poster: movie.poster_path
    })
  )
  return cleanMovies;
}