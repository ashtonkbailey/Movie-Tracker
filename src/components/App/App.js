import React, { Component } from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import '../../index.scss';
import * as clean from '../../utils/cleaner';
import * as actions from '../../actions/index';
import MovieDisplay from '../../containers/MovieDisplay/MovieDisplay';
import UserDisplay from '../UserDisplay/UserDisplay';
import UserForm from '../../containers/UserForm/UserForm';

class App extends Component {
  // remember to add in userComponent
  // if the user is logged in, display userComponent
  // if user is not logged in, display login/signup button
  
  async componentDidMount(){
    const movies = await clean.cleanMovies()
      this.props.addMovies(movies)
    }


  render() {
    // will need to add a default/error page to router

    return (
      <div className="App">
        <h1>
          Movie
          <br/>
          Tracker
        </h1>
        <UserDisplay />
        <Switch>
          <Route exact path='/' render={() => <MovieDisplay type="home" />} />
          <Route path='/favorites' render={() => <MovieDisplay type="favorites" />} />
          <Route path='/signup' render={() => <UserForm type="signup" />} />
          <Route path='/login' render={() => <UserForm type="login" />} />
        </Switch>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  movies: state.movies
})

const mapDispatchToProps = (dispatch) => ({
  addMovies: (movies)=>dispatch(actions.addMovies(movies))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
