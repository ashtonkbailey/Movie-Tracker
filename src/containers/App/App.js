import React, { Component } from 'react';
import { Route, Switch, withRouter, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import '../../index.scss';
import MovieDisplay from '../../containers/MovieDisplay/MovieDisplay';
import UserDisplay from '../../containers/UserDisplay/UserDisplay';
import UserForm from '../../containers/UserForm/UserForm';
import PropTypes from 'prop-types'
import * as clean from '../../utils/cleaner';
import * as actions from '../../actions/index';

export class App extends Component {

  async componentDidMount() {
    const movies = await clean.cleanMovies()
    this.props.addMovies(movies)
  }

  render() {
    return (
      <div className="App">
        <Link to="/">
          <h1>
            Movie
            <br/>
            Tracker
          </h1>
        </Link>
        <UserDisplay />
        <Switch>
          <Route exact path='/' render={() => <MovieDisplay type="home" />} />
          <Route path='/favorites' render={() => <MovieDisplay type="favorites" />} />
          <Route path='/signup' render={() => <UserForm type="signup" />} />
          <Route path='/login' render={() => <UserForm type="login" />} />
          <Route path='' render={() => 
            (<div>
              <h1>Oops! Couldn't find that page!</h1>
              <Link to='/'><button className="go-home">E.T. Phone Home</button></Link>
            </div>)
          } />
        </Switch>
      </div>
    );
  }
}

App.propTypes = {
  movies: PropTypes.array
}

App.defaultProps = {
  movies: []
}

export const mapStateToProps = (state) => ({
  movies: state.movies
})

export const mapDispatchToProps = (dispatch) => ({
  addMovies: (movies) => dispatch(actions.addMovies(movies))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
