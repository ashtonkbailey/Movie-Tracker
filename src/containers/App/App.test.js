import React from 'react';
import { shallow , mount} from 'enzyme';
import { mapStateToProps, mapDispatchToProps, App } from '../App/App';
import * as clean from '../../utils/cleaner'
import * as actions from '../../actions/index';
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import { MemoryRouter } from 'react-router-dom'

jest.mock('../../utils/cleaner')

describe('App', () => {

  describe('App Component', () => {
    it('should match the snapshot', () => {
      const wrapper = shallow(<App />)
      expect(wrapper).toMatchSnapshot()
    })

    it('should get movies when the component mounts', async () => {
      const mockAdd = jest.fn()
      const wrapper = shallow(<App addMovies={mockAdd} />)

      await wrapper.instance().componentDidMount()

      expect(mockAdd).toBeCalled()
    }) 
  })

  describe('routes', () => {
    /// Can add routes if we want to / have time but not needed if we dont
  })

  describe('mapStateToProps', () => {
    it('should return a props object with data from store', () => {
      const mockState = {
        movies: [{ title: 'Movie', rating: 4, description: 'stuff'}]
      }
      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(mockState)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should return a props object with a method addMovies', () => {
      const mockDispatch = jest.fn()
      const expected = actions.addMovies()

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addMovies()

      expect(mockDispatch).toBeCalledWith(expected)
    })
  })
})