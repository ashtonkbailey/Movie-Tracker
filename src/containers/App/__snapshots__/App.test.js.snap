// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    movies={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/"
        >
          <h1
            className="title"
          >
            Movie
            <br />
            Tracker
          </h1>
        </Link>,
        <Connect(UserDisplay) />,
        <Switch>
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/favorites"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/signup"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/login"
            render={[Function]}
          />
          <Route
            path=""
            render={[Function]}
          />
        </Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <h1
            className="title"
          >
            Movie
            <br />
            Tracker
          </h1>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Movie",
              <br />,
              "Tracker",
            ],
            "className": "title",
          },
          "ref": null,
          "rendered": Array [
            "Movie",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "Tracker",
          ],
          "type": "h1",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />,
            <Route
              exact={true}
              path="/favorites"
              render={[Function]}
            />,
            <Route
              exact={true}
              path="/signup"
              render={[Function]}
            />,
            <Route
              exact={true}
              path="/login"
              render={[Function]}
            />,
            <Route
              path=""
              render={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/favorites",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/signup",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/login",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/"
          >
            <h1
              className="title"
            >
              Movie
              <br />
              Tracker
            </h1>
          </Link>,
          <Connect(UserDisplay) />,
          <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/favorites"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/signup"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/login"
              render={[Function]}
            />
            <Route
              path=""
              render={[Function]}
            />
          </Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h1
              className="title"
            >
              Movie
              <br />
              Tracker
            </h1>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Movie",
                <br />,
                "Tracker",
              ],
              "className": "title",
            },
            "ref": null,
            "rendered": Array [
              "Movie",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "Tracker",
            ],
            "type": "h1",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />,
              <Route
                exact={true}
                path="/favorites"
                render={[Function]}
              />,
              <Route
                exact={true}
                path="/signup"
                render={[Function]}
              />,
              <Route
                exact={true}
                path="/login"
                render={[Function]}
              />,
              <Route
                path=""
                render={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/favorites",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/signup",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/login",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
